### PR TITLE
added multivariate dataset  with us consumption over time

### PR DESCRIPTION
**Pull request by SebasKoel**
Reference Issues/PRs
#376, #380, #382 

What does this implement/fix? Explain your changes.
added multivariate dataset so we can add support for multivariate/exogenous data of #382

### CODE REVIEW

File: sktime/datasets/base.py
Comment (mloning) [inline sktime/datasets/base.py:667]: Quarter is the time index, no? But predicting the time index doesn't make much sense. So it'd set Quarter as the index of the data, but given that we don't support period/datetime indices at the moment, I'd just drop it.

File: sktime/datasets/base.py
Comment (mloning) [inline sktime/datasets/base.py:670]: linting (see flake8) probably failed because there's an no empty line at the end

File: sktime/datasets/base.py
Comment (mloning) [inline sktime/datasets/base.py:619]: I'd call it `y_name`, please add it to the doc string 
```suggestion
def load_uschange(y_name="Consumption"):
```

**Review by mloning — COMMENTED** [review]
Thanks for the PR! Looks good in general, left three minor comments. Happy to merge afterwards!

File: sktime/datasets/base.py
Comment (SebasKoel) [reply sktime/datasets/base.py:667]: Yes Quarter is just a string. It's just being sorted on it as a double check (the csv is already sorted). It's being dropped. But it's also possible to get it back as y column, indeed it doesn't really make sense but else it should return an error when 'Quarter' is chosen which I don't see as better then having the option of having it returned as y

**Review by mloning — COMMENTED** [review]
There are a few pkl files on the PR now which we probably don't need, right?

File: sktime/datasets/tests/test_uschange.py
Comment (SebasKoel) [reply sktime/datasets/tests/test_uschange.py:15]: The pickle files are the X and y in pickle files and then it just compares these with the dataframe/serie returned by the function. Now the entire dataset is in it but for larger datasets I would've done i.e. the first 20 rows. I discussed this with Omar and he preferred not to have binary files, but just assert the column names, shapes and dtypes.  I think it's better this way because it checks the actual data as well, if someone accidentally changed the csv. But I don't mind changing it to what Omar suggested. Do you want me to change it?

Comment (mloning) [reply sktime/datasets/tests/test_uschange.py:15]: Ah okay, I think it's enough to have checks for what @big-o suggested, I don't want to have two copies of the data, the data itself is not that important I think, it's just toy data. Also someone may accidentally change the copy of the data, right? So a better test would be to download and compare the dataset from a reference source but I think this is not necessary. If you want to write unit tests for the loading function, it would be very useful to generalise them to test the other loading functions too (only the output object types, data types).

Comment (SebasKoel) [reply sktime/datasets/tests/test_uschange.py]: Okay just removed the pickle files and only use assertions. Also tried to make a general dataset testing function which can be used for other datasets as well.

File: sktime/datasets/tests/test_uschange.py
Comment (mloning) [inline sktime/datasets/tests/test_uschange.py:12]: This look good! The only thing I'd change is to replace `len` with `shape`.

File: sktime/datasets/tests/test_uschange.py
Comment (mloning) [inline sktime/datasets/tests/test_uschange.py:26]: I've only seen `assert` being used as a statement rather than function, please replace here and throughout, or is this a new thing? 
```suggestion
        assert isinstance(y, pd.Series)
```

File: sktime/datasets/tests/test_uschange.py
Comment (mloning) [inline sktime/datasets/tests/test_uschange.py:44]: Could you run this on `load_airline` too? You could use 
```python
@pytest.mark.parametrize("data_loader", [load_airline, load_uschange, ...])
def test_data_loaders(data_loader):
    ...
```
This will iterate over `data_loader` and run a separate test for each one.

Comment (mloning) [reply sktime/datasets/tests/test_uschange.py:44]: Of course we would have to iterate over the checks too ... perhaps we could add the `checks` dictionary to the docstring and retrieve it via `data_loader.__doc__`, just a thought

**Review by mloning — COMMENTED** [review]
Looks great! Left some minor comments and there's still this file: sktime/datasets/data/Uschange/Uschange.pkl

File: sktime/datasets/tests/test_uschange.py
Comment (big-o) [reply sktime/datasets/tests/test_uschange.py:44]: Maybe it would be cleaner to just declare the checks dictionary at the top of the module and then you would just be parametrising over the dictionary keys? e.g.:

```python
_CHECKS = {
    "uschange": {
        'loader': load_uschange,
        'len_y': 187,
        ...
    }
}

@pytest.mark.parametrize("dataset", [load_airline, load_uschange, ...])
def test_data_loaders(dataset):
    data_loader = _CHECKS[dataset]["loader"]
    ...
```

**Review by mloning — APPROVED** [review]
Looks good to me! @big-o I'll wait for your approval before we merge.

**Comment by big-o** [comment]
As mentioned in the above comment is be happy to approve now; the above change is very minor and could either be done in this PR or at a later date, if you think it's needed.

**Comment by mloning** [comment]
Hi @SebasKoel, could you confirm if you want to implement @big-o's suggestion? I wouldn't mind merging it now.

**Review by mloning — CHANGES_REQUESTED** [review]
Hi @SebasKoel, 

There's still the `Uschange.pkl` file in the PR and the changes to the `sktime/forecasting/compose/_reduce.py` which we wanted to have in a separate PR. 

I believe you should be able to remove them as follows:
* `git rm sktime/datasets/data/Uschange/Uschange.pkl`
* `git checkout master -- sktime/forecasting/compose/_reduce.py`
See if that works!

**Review by mloning — APPROVED** [review]

**Comment by mloning** [comment]
Thanks @SebasKoel! 🎉

**Review by big-o — COMMENTED** [review]
All looks good to me - as mentioned before I don't think the comment about the unit test really matters here, but maybe it should be considered when more unit tests are added in the future.

**Pull request merged**
